mergify_engine/actions/queue.py

```diff
@@ -228,6 +228,27 @@ async def run(
                     "The merge `method` or the queue configuration must be updated.",
                 )
 
+        protection = await ctxt.repository.get_branch_protection(
+            ctxt.pull["base"]["ref"]
+        )
+        if (
+            protection
+            and "strict" in protection["required_status_checks"]
+            and protection["required_status_checks"]["strict"]
+        ):
+            if self.queue_rule.config["batch_size"] > 1:
+                return check_api.Result(
+                    check_api.Conclusion.FAILURE,
+                    "batch_size > 1 is not compatible with branch protection a setting",
+                    "The branch protection setting `Require branches to be up to date before merging` must be unset.",
+                )
+            elif self.queue_rule.config["speculative_checks"] > 1:
+                return check_api.Result(
+                    check_api.Conclusion.FAILURE,
+                    "speculative_checks > 1 is not compatible with branch protection a setting",
+                    "The branch protection setting `Require branches to be up to date before merging` must be unset.",
```

Comment:
```suggestion
```suggestion
                    "Merge queue configuration is not compatible with one of the branch protection settings",
                    "The branch protection setting `Require branches to be up to date before merging` must be disabled when `speculative_checks > 1`.",
```
```